src/helpers/Authentication.test.spec.ts

```diff
@@ -0,0 +1,319 @@
+import { Test, TestingModule } from '@nestjs/testing'
+import { Authentication } from './Authentication'
+import { CACHE_MANAGER } from '@nestjs/cache-manager'
+import { Cache } from 'cache-manager'
+import { JwtService } from '@nestjs/jwt'
+import { Query } from './Query'
+import { Schema } from './Schema'
+import { Roles } from './Roles'
+import { RolePermission } from '../types/roles.types'
+import { Logger } from './Logger'
+import { ConfigService } from '@nestjs/config'
+import { AuthService } from '../app.service.auth'
+import { JwtModule } from '@nestjs/jwt'
+import { AuthLocation } from '../types/auth.types'
+import { LLANA_AUTH_TABLE } from '../app.constants'
+
+describe('Authentication Helper', () => {
+	let authentication: Authentication
+	let cacheManager: Cache
+	let jwtService: JwtService
+	let query: Query
+	let schema: Schema
+	let roles: Roles
+	let authService: AuthService
+	const cacheKey = 'auth:rules:JWT'
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [
+				JwtModule.register({
+					secret: 'test-secret-key', // Match the secret used in JwtService mock
+					signOptions: { expiresIn: '1h' },
+				}),
+			],
+			providers: [
+				Authentication,
+				{
+					provide: CACHE_MANAGER,
+					useValue: {
+						get: jest.fn(),
+						set: jest.fn(),
+						reset: jest.fn(),
+					},
+				},
+				{
+					provide: JwtService,
+					useValue: {
+						signAsync: jest.fn().mockImplementation(payload => {
+							const jwt = require('jsonwebtoken')
+							const secret = 'test-secret-key'
+							return Promise.resolve(
+								jwt.sign(payload, secret, {
+									expiresIn: '1h',
+								}),
+							)
+						}),
+						verifyAsync: jest.fn().mockImplementation(token => {
+							const jwt = require('jsonwebtoken')
+							const secret = 'test-secret-key'
+							try {
+								console.log('Verifying token with secret:', secret)
```

Comment:
You should not leave console.log testing commands in your code when you submit your PR, they should be removed beforehand.